pootle/apps/pootle_store/migrations/0027_unit_created_by_squashed_0055_fill_unit_source_data.py

```diff
@@ -0,0 +1,437 @@
+        ),
+        RemoveFieldIfExists(
+            model_name='unit',
+            name='source_wordcount',
+        ),
+        RemoveFieldIfExists(
+            model_name='unit',
+            name='commented_by',
+        ),
+        RemoveFieldIfExists(
+            model_name='unit',
+            name='commented_on',
+        ),
+        RemoveFieldIfExists(
+            model_name='unit',
+            name='reviewed_by',
+        ),
+        RemoveFieldIfExists(
+            model_name='unit',
+            name='reviewed_on',
+        ),
+        RemoveFieldIfExists(
+            model_name='unit',
+            name='submitted_by',
+        ),
+        RemoveFieldIfExists(
+            model_name='unit',
+            name='submitted_on',
+        ),
+        migrations.AlterField(
+            model_name='unitchange',
+            name='commented_by',
+            field=models.ForeignKey(null=True, on_delete=models.SET(pootle.core.user.get_system_user), related_name='commented', to=settings.AUTH_USER_MODEL),
+        ),
+        migrations.AlterField(
+            model_name='unitchange',
+            name='reviewed_by',
+            field=models.ForeignKey(null=True, on_delete=models.SET(pootle.core.user.get_system_user), related_name='reviewed', to=settings.AUTH_USER_MODEL),
+        ),
+        migrations.AlterField(
+            model_name='unitchange',
+            name='submitted_by',
+            field=models.ForeignKey(null=True, on_delete=models.SET(pootle.core.user.get_system_user), related_name='submitted', to=settings.AUTH_USER_MODEL),
+        ),
+        migrations.CreateModel(
+            name='SuggestionState',
+            fields=[
+                ('id', models.AutoField(auto_created=True, primary_key=True, serialize=False, verbose_name='ID')),
+                ('name', models.CharField(db_index=True, max_length=16)),
+            ],
+            options={
+                'abstract': False,
+                'db_table': 'pootle_store_suggestion_state',
+            },
+        ),
+        migrations.RunPython(add_default_suggestion_states),
+        migrations.AddField(
+            model_name='suggestion',
+            name='tmp_state',
+            field=models.CharField(choices=[('pending', 'Pending'), ('accepted', 'Accepted'), ('rejected', 'Rejected')], db_index=False, default='pending', max_length=16),
+        ),
+        migrations.RunPython(set_suggestion_tmp),
+        migrations.RemoveField(
+            model_name='suggestion',
+            name='state',
+        ),
+        migrations.AddField(
+            model_name='suggestion',
+            name='state',
+            field=models.ForeignKey(null=True, on_delete=django.db.models.deletion.SET_NULL, related_name='suggestions', to='pootle_store.SuggestionState'),
+        ),
+        migrations.RunPython(set_suggestion_states),
+        migrations.RemoveField(
+            model_name='suggestion',
+            name='tmp_state',
+        ),
```

Comment:
the only difference performance wise is storing the fk - but to be a performance boost i think it would need to be stored as an int not fk